Tools/scripts/summarize_stats.py

```diff
@@ -2,908 +2,988 @@
+            for key, value in opcode_stat.items():
+                if key.startswith("pair_count"):
+                    name_j, _, _ = key[len("pair_count") + 1 :].partition("]")
+                    if value:
+                        pair_counts.append((value, (name_i, name_j)))
+        pair_counts.sort(reverse=True)
+        return pair_counts
+
+    def get_total(self, prefix: str) -> int:
+        return sum(
+            x.get("execution_count", 0) for x in self.get_opcode_stats(prefix).values()
         )
-        base_rows = calculate_specialization_success_failure(base_family_stats)
-        head_rows = calculate_specialization_success_failure(head_family_stats)
-        rows = join_rows(base_rows, head_rows)
-        if rows:
-            print_title("Specialization attempts", 4)
-            emit_table(
-                ("", "Base Count:", "Base Ratio:", "Head Count:", "Head Ratio:"), rows
-            )
-            base_rows = calculate_specialization_failure_kinds(
-                name, base_family_stats, defines
-            )
-            head_rows = calculate_specialization_failure_kinds(
-                name, head_family_stats, defines
-            )
-            emit_table(
-                (
-                    "Failure kind",
-                    "Base Count:",
-                    "Base Ratio:",
-                    "Head Count:",
-                    "Head Ratio:",
-                ),
-                join_rows(base_rows, head_rows),
-            )
 
 
-def gather_stats(input):
-    # Note the output of this function must be JSON-serializable
-
-    if os.path.isfile(input):
-        with open(input, "r") as fd:
-            stats = json.load(fd)
-
-        stats["_stats_defines"] = {
-            int(k): v for k, v in stats["_stats_defines"].items()
-        }
-        stats["_defines"] = {int(k): v for k, v in stats["_defines"].items()}
-        return stats
-
-    elif os.path.isdir(input):
-        stats = collections.Counter()
-        for filename in os.listdir(input):
-            with open(os.path.join(input, filename)) as fd:
-                for line in fd:
-                    try:
-                        key, value = line.split(":")
-                    except ValueError:
-                        print(
-                            f"Unparsable line: '{line.strip()}' in  {filename}",
-                            file=sys.stderr,
-                        )
-                        continue
-                    key = key.strip()
-                    value = int(value)
-                    stats[key] += value
-            stats["__nfiles__"] += 1
+class Count(int):
+    def markdown(self) -> str:
+        return format(self, ",d")
 
-        import opcode
 
-        stats["_specialized_instructions"] = [
-            op for op in opcode._specialized_opmap.keys() if "__" not in op
-        ]
-        stats["_stats_defines"] = get_stats_defines()
-        stats["_defines"] = get_defines()
+class CountPer:
```

Comment:
Because my secret plan is to also introduce CSV output, where we would want to format this differently.

Reply:
And also a Ratio is rendered as a percentage, a CountPer is rendered as an integer.  "Uops run per trace" is much better represented as an integer rather than a percentage.

Reply:
I think uops per trace is more naturally a float than an int. I agree that 30[.0] is better than 3000% though.
Maybe add a `percent: bool=True` argument to `Ratio`'s `__init__`?